examples/nested/run-tests.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+set -xe
+
+SCRIPT="$(readlink -f "$0")"
+SCRIPTPATH="$(dirname "${SCRIPT}")"
+TESTDIR="${SCRIPTPATH}/../../.github/tests"
+DEPS="${TESTDIR}/dependencies"
+
+# shellcheck source=/dev/null
+source "${SCRIPTPATH}/../../.github/scripts/parse-versions.sh"
+# shellcheck source=/dev/null
+source "${TESTDIR}/common.sh"
+
+helm_install=(helm upgrade --install --create-namespace)
+ns=spire-server
+
+teardown() {
+  helm uninstall --namespace "${ns}" spire 2>/dev/null || true
+  kubectl delete ns "${ns}" 2>/dev/null || true
+
+  helm uninstall --namespace mysql spire-root-server 2>/dev/null || true
+  kubectl delete ns spire-root-server 2>/dev/null || true
+}
+
+trap 'trap - SIGTERM && teardown' SIGINT SIGTERM EXIT
+
+kubectl create namespace spire-system 2>/dev/null || true
+kubectl label namespace spire-system pod-security.kubernetes.io/enforce=privileged || true
+kubectl create namespace "${ns}" 2>/dev/null || true
+kubectl label namespace "${ns}" pod-security.kubernetes.io/enforce=restricted || true
+
+"${helm_install[@]}" spire charts/spire \
+  --namespace spire-root-server \
+  --values "${DEPS}/spire-root-server-values.yaml" \
+  --wait
+
+kubectl get nodes -o go-template='{{range .items}}{{printf "%s\n" .metadata.uid}}{{end}}' | while read -r line; do
+  kubectl exec -t spire-server-0 -n "spire-root-server" -- spire-server entry create -spiffeID spiffe://example.org/example-cluster/nested-spire -parentID "spiffe://example.org/spire/agent/k8s_psat/example-cluster/$line" -selector k8s:pod-label:app.kubernetes.io/name:server -downstream
+done
+
+"${helm_install[@]}" --namespace "${ns}" --values "${SCRIPTPATH}/values.yaml,${SCRIPTPATH}/../production/values.yaml" \
```

Comment:
De-template this.  It makes it unclear that this is a helm upgrade (upsert) command, and makes it doubly unclear that this is the child cluster.  

In addition, de-template the namespace "ns" parameter as we are dealing with two namespaces.  If it must be held in a variable, it should be "child_namespace".  Additionally, if we are holding the namespace in a variable for the child, we MUST hold the parent namespace in a "parent_namespace" variable to match, and ideally would set both of those constants together near the top of the file.

I'm not 100% convinced that the need to template the namespace is by holding it in a variable is valuable; as the decision to do so creates issues, such as the issue of whether we wish to create a test that has two child clusters, in which case child_namespace might better be an array, and that then raises the question of how to make multiple arrays coordinate (Associate arrays with the cluster name as the key?  Indexed arrays?)

Reply:
This is a common pattern throughout the chart. I actually mostly just copied this file and tweaked a few things. We probably need an issue to fix these sorts of things more globally. that being said, I'll fix this one as you request.

Reply:
Done.